tests/allocator_api.rs

```diff
@@ -1,4 +1,4 @@
-#![feature(allocator_api)]
+#![feature(allocator_api, slice_ptr_get)]
```

Comment:
Could you avoid the use of new nightly features here? It just makes things harder when we have to update unrelated tests because the feature has changed in some breaking way since the last PR was made. I know that the tests will be slightly more verbose, but it is worth it to avoid forcing churn upon unrelated PRs.